apps/reply/lib.js

```diff
@@ -46,11 +46,20 @@ exports.reply = function (options) {
         options.fileOverride || "replies.json",
         true
       ) || [];
+
     replies.forEach((reply) => {
-      menu = Object.defineProperty(menu, reply.text, {
+      var displayString = reply.disp ?? reply.text;
+      var truncatedDisplayString = g.wrapString(displayString, 120);
+      // Generally handles truncating nicely, but falls down in long runs of emoji since they count as one image
+      if (truncatedDisplayString.length > 1) {
+        truncatedDisplayString = truncatedDisplayString.slice(0,1);
+        truncatedDisplayString[truncatedDisplayString.length-1] += "...";
+      }
+      menu = Object.defineProperty(menu, truncatedDisplayString, {

```

Comment:
`truncatedDisplayString` is an array isn't it? presumably you should be doing `truncatedDisplayString.join("\n")`?

Reply:
ping @WhelanB

Reply:
Sorry for the lack of updates on this PR - I was away for a few weeks without access to my PC, and my Bangle isn't functioning correctly atm. I think I was trying a few different ways of truncating here, and ended up with some leftover code - my intention was to just take ``truncatedDisplayString[0] + "..."`` as a hacky workaround for some issues I was seeing with showMenu's ability to handle some strings

for example, the following from Gadgetbridge:

``
this is a looooooooooooooooooooooooooooooooooong message ❤️❤️❤️❤️
``

renders by default like this:
![image](https://github.com/user-attachments/assets/6b57158c-4fe0-4d25-92a5-e2453c6ab164)


vs by taking the first line and adding the ellipsis:
![image](https://github.com/user-attachments/assets/a6d3a7c6-1bdd-41ab-84db-13e21f651747)


Is there a recommended way to handle this kind of overflow? if not I'll update the code to just use ``truncatedDisplayString[0] + "..."``

Reply:
I get what you're trying to do - but my point was that `g.wrapString` returns an Array, not a string, so truncatedDisplayString is an Array. You can't define a property on an object that's an array, so Espruino converts it back to a string with commas.

```
a = []; a[[1,2,3]] = 4
=[ "1,2,3": 4 ]
```

In your case it works because you ensure it's a single-element array so no commas are added, but if the linter gets better it'll probably pick this up as an issue, and either way it's a bit confusing and if someone tries to modify it, they'll almost certainly break it. 

You could just do:

```
menu = Object.defineProperty(menu, truncatedDisplayString[0], {
```

But it felt quicker for me to make a change to fix this myself: https://github.com/espruino/BangleApps/pull/3556/commits/648ebd432132867079c6cb44f5077806ead5bd2c

Maybe you could just check that it still works ok after that?

Reply:
All looks good here - I've also opened the associated PR against Gadgetbridge on Codeberg:
https://codeberg.org/Freeyourgadget/Gadgetbridge/pulls/4182